### DISCUSSION TITLE
How to cancel an action using a plugin?

### DISCUSSION THREAD

**Discussion by ikar1234** [vuejs/pinia#1945] [Help and Questions] [answered]
In the [very first paragraph in the plugin docs](https://pinia.vuejs.org/core-concepts/plugins.html), a bullet point says that we can change or even cancel actions. Can anyone provide an example of how this is achievable?

Right now, I'm using a workaround by setting some property using `$onAction`, then using that property inside the action to decide whether to manually cancel the action. It feels a bit hacky to me, and it certainly does not provide the best interface in the store (since I have to use the custom property inside the actions).

**Comment by posva** [comment, marked as answer]
This was a leftover from the past, it was removed due to the conflict of having multiple `$onAction()` not being able to decide whether an action should be canceled or not (https://github.com/vuejs/pinia/commit/4dc1f1bec8f3db508c02fa6bdcc8203f280cbb3e).


Technically, one should be able to add the property in a plugin (similar to devtools plugin wrapping each action) and do the check within the wrapped action to completely skip calling the action. That would prevent you from changing actions at all.

I updated the docs at 5bc44d3 (in the docs branch).
